### ISSUE TITLE
Wrong resolved import path

### ISSUE THREAD

**Issue by victordidenko** [simonhaenisch/rollup-plugin-typescript-paths#10]
This kind of path (without any `@` sign):
```json
"paths": {
  "core/*": ["src/core/*"],
```

And import like this (could be auto-imported by Visual Studio Code)
```ts
import { smth } from '../../../core/functions'
```

In conjunction will result in a wrong resolved path.

What if add caret in RegExp?
https://github.com/simonhaenisch/rollup-plugin-typescript-paths/blob/5f6bbc85ebde45dcd9be36f0b4f4ba169388e1ff/index.ts#L21
like this:
```ts
new RegExp('^' + path.replace('*', '\\w*')).test(importee),
```

I'm not sure if this correct fix or no, though...

**Comment by victordidenko** [comment]
I mean, adding caret definitely fixes issue for me, but I don't know, will this be correct for any possible path in TypeScript configuration

**Comment by simonhaenisch** [comment]
Hey I haven't worked on this for a while but now that i'm looking at it, it doesn't seem to make too much sense to replace `*` with `\w*` because that would also break if there's a digit in the path 🙈 (which probably has just gone unnoticed because it's relatively rare that one would do this).

Anyway, in your case, if you're importing it as

```ts
import { smth } from 'core/functions';
```

then it should know how to resolve it (that's how the plugin is supposed to work, at least). Is that how you're writing the import, or are you using the auto-imported version from your example?

**Comment by victordidenko** [comment]
Import like this
```js
import { smth } from 'core/functions';
```
works just fine, yes.

But there are more people in team, than only me, and someone can write `smth` in Visual Studio Code, then it will show this nice message, meaning "Unknown identifier, let me auto-import it for you", and after accepting this proposal, Visual Stido Code will insert line
```js
import { smth } from '../../../core/functions';
```
in the header.

This import works fine in dev environment (storybook, built by it's internal Webpack with plugin `tsconfig-paths-webpack-plugin`), so no one will suspect any trouble.

Until CI will try to build project, using Rollup and `rollup-plugin-typescript-paths`.

**Comment by simonhaenisch** [comment]
Ok, I see... yeah both paths should work, and I think I see what you mean with the caret now. Makes sense to add it IMO, I'll add some tests to make sure nothing breaks.

Thanks for raising the issue!

Side note: you can control how VS Code auto-imports with the `typescript.preferences.importModuleSpecifier`:

```js
{
  // Preferred path style for auto imports.
  //  - shortest: Prefers a non-relative import only if one is available that has fewer path segments than a relative import.
  //  - relative: Prefers a relative path to the imported file location.
  //  - non-relative: Prefers a non-relative import based on the `baseUrl` or `paths` configured in your `jsconfig.json` / `tsconfig.json`.
  //  - project-relative: Prefers a non-relative import only if the relative import path would leave the package or project directory. Requires using TypeScript 4.2+ in the workspace.
  "typescript.preferences.importModuleSpecifier": "shortest",
}
```

**Comment by simonhaenisch** [comment]
Sorry took me a while to actually get this fixed, but it's done now! I've published it as `1.3.1`, should be available already.

Thanks again for raising this!

**Comment by victordidenko** [comment]
Thank you!
